configuration/src/main/java/io/airlift/configuration/resolver/ConfigurationResolverManager.java

```diff
@@ -0,0 +1,200 @@
+import java.nio.file.DirectoryStream;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Streams.stream;
+import static com.google.common.io.Files.getNameWithoutExtension;
+import static io.airlift.configuration.ConfigurationLoader.loadPropertiesFrom;
+import static java.nio.file.Files.newDirectoryStream;
+import static java.util.Locale.ENGLISH;
+
+public final class ConfigurationResolverManager
+{
+    private static final ImmutableList<String> SPI_PACKAGES = ImmutableList.<String>builder()
+            .add("io.airlift.spi.")
+            .add("com.fasterxml.jackson.annotation.")
+            .add("io.airlift.slice.")
+            .add("org.openjdk.jol.")
+            .build();
+
+    private static final Logger log = Logger.get(ConfigurationResolverManager.class);
+
+    private static final String NAME_PROPERTY = "configuration-provider.name";
+
+    private static final Pattern CONFIG_PRODIVER_NAME_PATTERN = Pattern.compile("[a-zA-Z][a-zA-Z0-9_-]*");
+
+    private final Map<String, ConfigurationValueResolverFactory> configurationProviderFactories = new ConcurrentHashMap<>();
+    private final AtomicReference<Map<String, ConfigurationValueResolver>> configurationProviders = new AtomicReference<>(ImmutableMap.of());
+    private final File installedConfigurationResolver;
+    private final List<File> configFiles;
+
+    public ConfigurationResolverManager(Map<String, String> configurationResolverConfig)
+    {
+        ConfigurationFactory configurationFactory = new ConfigurationFactory(configurationResolverConfig);
+        ConfigurationResolverConfig config = configurationFactory.build(ConfigurationResolverConfig.class);
+        this.installedConfigurationResolver = config.getInstalledConfigurationResolver();
+        configFiles = ImmutableList.copyOf(config.getConfigurationResolverFiles());
+    }
+
+    public void load()
+    {
+        listFiles(installedConfigurationResolver).stream()
+                .filter(File::isDirectory)
+                .forEach(file -> loadConfigurationResolvers(file.getAbsolutePath(), () -> createClassLoader(file.getName(), buildClassPath(file))));
+        ImmutableMap.Builder<String, ConfigurationValueResolver> builder = ImmutableMap.builder();
+        for (File configFile : configFiles) {
+            builder.put(getNameWithoutExtension(configFile.getName()), loadConfigProvider(configFile.getAbsoluteFile()));
+        }
+        builder.put("env", new EnvironmentVariableConfigurationValueResolver());
```

Comment:
We should consider making `env` be a configurable plugin, just like any other plugin. Some people may want to disable it for security purposes. Or if you're using a different provider, you might not want or need environment variables.

Reply:
But won't it contradict with this comment - https://github.com/airlift/airlift/pull/1217#discussion_r1672323394. We could migrate them to `configuration-resolvers` plugin which captures a set of predefined `ConfigPlugin` -

Reply:
But would it be backward compatible ? Like in order to use the `env` we need to add an entry to the `config-resolver.properties` - Won't it break for the existing Trino deployment